domain/lib/src/usecases/offline_mode/auto_sync_available_offline_document_interactor.dart

```diff
@@ -43,21 +43,31 @@ class AutoSyncAvailableOfflineDocumentInteractor {
 
   AutoSyncAvailableOfflineDocumentInteractor(this._documentRepository, this._tokenRepository, this._credentialRepository);
 
-  Future<Either<Failure, Success>> execute(Document document) async {
+  Future<Either<Failure, Success>> execute(Document? document) async {
     try {
-      final downloadPreviewType = document.mediaType.isImageFile()
-        ? DownloadPreviewType.image
-        : DownloadPreviewType.original;
+      if (document == null) {
+        return Left<Failure, Success>(MakeAvailableOfflineDocumentFailure(OfflineModeActionResult.failure));
+      }
+
+      final documentLocal = await _documentRepository.getDocumentOffline(document.documentId);
+      var filePath = documentLocal?.localPath ?? '';
+      final shaLocal = documentLocal?.sha256sum;
 
-      final filePath = await Future.wait([_tokenRepository.getToken(), _credentialRepository.getBaseUrl()], eagerError: true)
-          .then((List responses) async {
-        return await _documentRepository.downloadMakeOfflineDocument(
-          document.documentId,
-          document.name,
-          downloadPreviewType,
-          responses[0],
-          responses[1]);
-      });
+      if (shaLocal != document.sha256sum) {
+        final downloadPreviewType = document.mediaType.isImageFile()
+            ? DownloadPreviewType.image
+            : DownloadPreviewType.original;
+
+        filePath = await Future.wait([_tokenRepository.getToken(), _credentialRepository.getBaseUrl()], eagerError: true)
+            .then((List responses) async {
+          return await _documentRepository.downloadMakeOfflineDocument(
+              document.documentId,
+              document.name,
+              downloadPreviewType,
+              responses[0],
+              responses[1]);
```

Comment:
please define this variable explicit: `final name = response[0]` for more readable